### ISSUE TITLE
SAM CLI 1.33 Cannot Resolve ECR Container Images (Gets 404)

### ISSUE THREAD

**Issue by curtis-trynow-io** [aws/aws-sam-cli#3329]
### Description:

After upgrading from 1.32 to 1.33 both locally and in CI/CD pipelines, our `sam build --use-container` processes are failing with:

```
samcli.local.docker.manager.DockerImagePullFailedException: Could not find public.ecr.aws/sam/build-python3.8:latest-x86_64 image locally and failed to pull it from docker.
```

Indeed, this image does not exist, and I suspect it is because of the architecture suffix `-x86_64` which previously was not used.  If you use just `latest`, it is fine, but adding that results in nothing being found.

### Steps to reproduce:

Make a new directory, add a template file with:

```
AWSTemplateFormatVersion: 2010-09-09
Transform:
  - AWS::Serverless-2016-10-31
Resources:
  someFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: ./src
      Handler: handler.handler
      MemorySize: 128
      Runtime: python3.8
      Timeout: 5
```

and run `sam build --use-container`.  I doubt you even need `CoreUri` but rushing to get this submitted.  It really just needs to be something to pull the Python build image.

### Observed result:

https://pastebin.com/raw/vPFNxjiD

I don't think you need debug output for this one.  The image does not exist.  You can try pulling it with a plain:

```
docker pull public.ecr.aws/sam/build-python3.8:latest-x86_64
```

### Expected result:

The image exists and can be pulled.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS but also happens in Linux docker-in-docker containers
2. `sam --version`: 1.33
3. AWS region: not specific

`Add --debug flag to command you are running`

**Comment by curtis-trynow-io** [comment]
Happens for other runtimes too, like `nodejs14.x`:

```
samcli.local.docker.manager.DockerImagePullFailedException: Could not find public.ecr.aws/sam/build-nodejs14.x:latest-x86_64 image locally and failed to pull it from docker.
```

Trivially reproducible by just adjusting the runtime in the template YAML.

**Comment by mgrandis** [comment]
Sorry we haven't been able to publish the images yet due to an issue.
Please rollback to 1.32.0 for now for container build.

**Comment by curtis-trynow-io** [comment]
I mean that's great but that would presumably waste hundreds to thousands of hours of engineering time to implement across all orgs that use the SAM CLI versus you all just re-publishing your images under the new tags (can overwrite later), removing the 1.33 release, or re-publishing the 1.32 release as 1.34 so it gets picked up automatically by everyone's CI/CD chains, apps, Homebrew, package maintainer repos, etc.  All of those options would take enormously less time and fix it for everyone.

**Comment by mgrandis** [comment]
Got it, sorry.
We are doing our best to publish the images now.

**Comment by nickjj** [comment]
I'm using Sam 1.18.2 and I'm receiving the same type of Python error. This only started happening sometime on September 30th 2021 as far as I know.

My CI logs say: `SAM CLI update available (1.33.0); (1.18.2 installed)` and it throws `Error: Could not find amazon/aws-sam-cli-emulation-image-python3.8:rapid-1.18.2 image locally and failed to pull it from docker`.

Is this something that will self resolve?

**Comment by c2tarun** [comment]
@curtis-trynow-io the images has been published now and I can confirm that container build is working with sam-cli 1.33.0

```
❯ sam --version && sam build --use-container
SAM CLI, version 1.33.0
Starting Build inside a container
Building codeuri: /workplace/issues/sam-cli/sam-app/hello_world runtime: python3.8 metadata: {} architecture: x86_64 functions: ['HelloWorldFunction']

Fetching public.ecr.aws/sam/build-python3.8:latest-x86_64 Docker container image......
Mounting /workplace/issues/sam-cli/sam-app/hello_world as /tmp/samcli/source:ro,delegated inside runtime container

Build Succeeded

Built Artifacts  : .aws-sam/build
Built Template   : .aws-sam/build/template.yaml

Commands you can use next
=========================
[*] Invoke Function: sam local invoke
[*] Deploy: sam deploy --guided

Running PythonPipBuilder:ResolveDependencies
Running PythonPipBuilder:CopySource
```

@nickjj, Can you upgrade to version 1.33.0 and see if it resolves your issue? If the issue persists for you please feel free to open a new issue as that would be a different problem then the one in this issue.

Thanks
